src/pages/resultsView/download/CaseAlterationTable.tsx

```diff
@@ -197,54 +204,49 @@ export default class CaseAlterationTable extends React.Component<ICaseAlteration
             }
         ];
 
-        const geneSet : {[x: string]: OQLLineFilterOutput<AnnotatedExtendedAlteration>} = {};
-        this.props.oqls.forEach(oql => {
-            if (oql.gene in geneSet) {
-                geneSet[oql.gene].oql_line += oql.oql_line;
-            }
-            else {
-                geneSet[oql.gene] = oql;
-            }
-        });
-        
-        _.forEach(geneSet, (oql) => {
-            const alterationTypes = this.props.alterationTypes;
-            //add column for each gene
+        // track columns
+        _.forEach(this.props.trackLabels, (trackLabel) => {
+            // add column for each track
             columns.push({
-                name: `${oql.gene}`,
-                tooltip: <span>{oql.oql_line}</span>,
-                headerDownload: (name: string) => `${oql.gene}`,
+                name: `${trackLabel}`,
+                tooltip: <span>{trackLabel}</span>,
+                headerDownload: (name: string) => `${trackLabel}`,
                 render: (data: ICaseAlteration) => {
-                    const pseudoOqlSummary = getPseudoOqlSummaryByAlterationTypes(data.oqlDataByGene, oql.gene, alterationTypes); 
-
+                    const pseudoOqlSummary = getPseudoOqlSummaryByAlterationTypes(data.oqlData, trackLabel, this.props.alterationTypes); 
                     return <span className={pseudoOqlSummary.summaryClass}>{pseudoOqlSummary.summaryContent}</span>;
                 },
-                download: (data: ICaseAlteration) => getPseudoOqlSummaryByAlterationTypes(data.oqlDataByGene, oql.gene, alterationTypes)!.summaryContent,
-                sortBy: (data: ICaseAlteration) => getPseudoOqlSummaryByAlterationTypes(data.oqlDataByGene, oql.gene, alterationTypes)!.summaryContent,
+                download: (data: ICaseAlteration) => getPseudoOqlSummaryByAlterationTypes(data.oqlData, trackLabel, this.props.alterationTypes)!.summaryContent,
+                sortBy: (data: ICaseAlteration) => getPseudoOqlSummaryByAlterationTypes(data.oqlData, trackLabel, this.props.alterationTypes)!.summaryContent,
                 filter: (data: ICaseAlteration, filterString: string, filterStringUpper: string) =>
```

Comment:
rather than mainpulating the casing of things, it's better to use case insensitive regexp.  
` (new RegExp(/ab+c/, 'i')).test(someStr)`

Reply:
i realize this is is code taken from elsewhere.  just want to make sure we do it correctly going forward